src/NHibernate/SqlCommand/SqlString.cs

```diff
@@ -404,6 +404,11 @@ public int IndexOfCaseInsensitive(string text)
 			return IndexOf(text, 0, _length, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
It appears to me this one should be switched to `StringComparison.OrdinalIgnoreCase`. Invariant does some Unicode character equivalency logic, which is most likely to cause "false positives" from a query standpoint. Moreover case insensitive ordinal is more performant than invariant. And this method is used either on ASCII characters or on query aliases.

But overall they are more than hundred cases were invariant is used in the code base. It looks to me most of them if not all should be switched to ordinal. Maybe worth a dedicated PR.

(By the way if changing this case here, the `text` parameter comment looks obsolete, its last part should be removed.)

Reply:
>It appears to me this one should be switched to StringComparison.OrdinalIgnoreCase. 

Yeah. But it's public method. I'm afraid to change its behaviour :) 

> It looks to me most of them if not all should be switched to ordinal. 
Agreed (but not as part of this PR)

Reply:
> Yeah. But it's public method. I'm afraid to change its behaviour :)

Fixing a behavior which is a bug is OK. I do not think the invariant behavior discrepancies with ordinal can be anything else than bugs for this case, but well, maybe I am wrong. Something that is equal with invariant but not equal with ordinal looks very likely to me to be considered not equal by the database query parser as well, in which case we would have a bug. Still this can be the subject of another PR, switching all invariant which should be to ordinal. I also think the change of most of them could be considered as also fixing bug possibilities.

@hazzik, what is your view on this subject? Do you think it is ok to switch from invariant to ordinal in a minor release, considering it is also a bug fix, or would you rather have such change in a major release?